### ISSUE TITLE
Do not close the tag LI and nested list

### ISSUE THREAD

**Issue by FJCrux** [evbacher/gd2md-html#62]
Here is 2 problems:
- Only last <li> have </li>
- Nested list not in <li></li>

**Comment by Kainkainkain** [comment]
I have the same issue, quick solution is paste in vscode an replace open tag with </li><li> and then replace <ul></li> with just <ul> and then replace all <li></ul> with </ul>. With ignoring white space.

**Comment by pencerw** [comment]
i have this issue as well. it is indeed easy to fix in a text editor, but it would be great to have it baked into the plugin!

**Comment by pencerw** [comment]
i'm guessing, but is the issue here? https://github.com/evbacher/gd2md-html/blob/9b2ded3d5366858caa722966ee415825e144d306/addon/gdc.gs#L1167

**Comment by joshuauyi** [comment]
I experienced this same issue. Only the last li closing tag is rendered.

**Comment by evbacher** [comment]
I believe this is fixed with recent changes. Please let us know if you see evidence of this bug still happening.

**Comment by joshuauyi** [comment]
Open and closing tags were rendered when I tested, thanks for fixing this!

NB: the `Nested list not in <li></li>` mentioned by the original post still exists

**Comment by evbacher** [comment]
Can any of you link a doc (view permission is fine) that illustrates the nested list issue. I'm not following exactly what the problem is (and I don't think I have any test case that shows the problem yet).

Thanks!

**Comment by joshuauyi** [comment]
You can see a sample document here https://docs.google.com/document/d/1eRw82Tjnl_g8RKUezszJfP83xI5oryTMKnCb05Ou-ec 
This renders this

```
<p>
Nested List
</p>
<ul>
<li>Item 1</li>
<li>Item 2</li> 
<ul>
<li>Sub Item 1</li>
<li>Sub Item 2</li> 
</ul>
<li>Item 3</li>
</ul>
```

instead of this, with the sub ul inside a li tag

```
<p>
Nested List
</p>
<ul>
<li>Item 1</li>
<li>Item 2</li> 
<li>
<ul>
<li>Sub Item 1</li>
<li>Sub Item 2</li> 
</ul>
</li> 
<li>Item 3</li>
</ul>
```

**Comment by evbacher** [comment]
Thanks for pointing this out, @joshuauyi -- we're getting down to fine points of HTML now!

Are you saying that the rendered HTML not rendering correctly? If so, that is a significant bug. But both versions seem to render fine in my limited tests.

Technically, you're correct: an unordered list cannot be the direct child of another unordered list. But you don't really need to create another list item.

If I do get around to fixing this, I'll make the nested list a child of the Item 2 list item, like this:

```
<p>
  Nested List
</p>
<ul>
  <li>Item 1</li>
  <li>
    Item 2
    <ul>
      <li>Sub Item 1</li>
      <li>Sub Item 2</li>
    </ul>
  </li>
  <li>Item 3</li>
</ul>
```

**Comment by joshuauyi** [comment]
Yes, they render fine, It's just the valid HTML, it still looks fine like you said.
